### PR TITLE
Multi input gates

### PR DESCRIPTION
**Pull request by Rmano**
First try at multi-input gates by John Kormylo. To be refined.

### CODE REVIEW

File: tex/pgfcirctripoles.tex [hunk 1]
```diff
@@ -136,421 +136,360 @@
...
-	  
-	  }
-	}
-}
 
+%%% added 
+
+\newcount{\pgf@circ@res@count}% reserve global register
+
+\newcommand{\pgf@circ@logicport@input}[1]% #1 = \pgfmathcounter
```
Comment (TheTeXnician) [inline tex/pgfcirctripoles.tex]: Please do not use `\newcommand` as `circuitikz` is also a ConTeXt module (I'll add a ConTeXt MWE in #105). Use TeX instead.

```suggestion
\def\pgf@circ@logicport@input#1% #1 = \pgfmathcounter
```

File: tex/pgfcirctripoles.tex [hunk 2]
```diff
@@ -136,421 +136,360 @@
...
-			
-			#2
-	  
-	  }
-	}
-}
 
+%%% added 
+
+\newcount{\pgf@circ@res@count}% reserve global register
```
Comment (TheTeXnician) [inline tex/pgfcirctripoles.tex]: At TeX level there are usually no braces around a `\newcount` (this is different from `\newcounter`, see e.g. https://www.texdev.net/2009/11/17/tex-counts-and-latex-counters/)

```suggestion
\newcount\pgf@circ@res@count% reserve global register
```

File: tex/pgfcirctripoles.tex [hunk 3]
```diff
@@ -136,421 +136,360 @@
...
+      \ifnum\pgf@circ@res@count<2 \pgf@circ@res@count=2\fi
+      \ifnum\pgf@circ@res@count>16 \pgf@circ@res@count=16\fi
+      \def\inputs{\the\pgf@circ@res@count}%
+    }%
+    \savedanchor\step{% 1/2 gap at edges
+      \pgf@circ@res@step = \pgfkeysvalueof{/tikz/circuitikz/tripoles/american #1 port/height}\pgf@circ@Rlen
+      \divide\pgf@circ@res@step by \pgf@circ@res@count
+      \pgfpoint{\pgf@circ@res@left}{\dimexpr\pgf@circ@res@up+0.5\pgf@circ@res@step}%
+    }%
+      \savedanchor\northeast{\pgfpoint{\pgf@circ@res@right}{\pgf@circ@res@up}}
```
Comment (TheTeXnician) [inline tex/pgfcirctripoles.tex]: Probably spaces should be equal in all lines…

```suggestion
    \savedanchor\northeast{\pgfpoint{\pgf@circ@res@right}{\pgf@circ@res@up}}
```

File: tex/pgfcirctripoles.tex [hunk 4]
```diff
@@ -865,151 +804,144 @@
 	}
 
 
-
+%%% start european logic ports (do really someone use them?)
```
Comment (TheTeXnician) [inline tex/pgfcirctripoles.tex]: As there are European users out there that actually use the European ports (like me) I would like to argue that the second half of the comment is unnecessary.

```suggestion
%%% start european logic ports
```

File: tex/pgfcirctripoles.tex (hunk 3 above)
Comment (Rmano) [reply tex/pgfcirctripoles.tex]: Yes, I have an issue open for a big clean-up of spaces... accepted, of course.

**Comment by Rmano** [comment]
Reorganized the main commits. I still need to do doc work.

**Comment by Rmano** [comment]
@TheTeXnician  --- what do you think?

**Comment by Rmano** [comment]
Sorry, closed and reopened by error.

**Comment by TheTeXnician** [comment]
Works for me, so it should be good to go :slightly_smiling_face:

**Pull request merged**